packages/story-editor/src/app/story/useStoryReducer/reducers/arrangeElement.js

```diff
@@ -86,38 +89,21 @@ function arrangeElement(state, { elementId, position, groupId = false }) {
   });
 
   const currentGroupId = page.elements[currentPosition].groupId;
+
   // If it's already there, do nothing.
   if (currentPosition === newPosition && currentGroupId === groupId) {
-    return state;
+    return;
   }
 
-  const newPageElements = [...page.elements];
-  newPageElements[currentPosition] = {
-    ...newPageElements[currentPosition],
-  };
-  if (groupId !== false) {
-    newPageElements[currentPosition].groupId = groupId;
+  // Update group id on current element
+  if (groupId) {
+    page.elements[currentPosition].groupId = groupId;
+  } else {
+    delete page.elements[currentPosition].groupId;
```

Comment:
Right now the `groupId` can be either a `string` = actual groupId (move into the group), `null/undefined` = remove the groupId (move out of the group) or `false` = don't update the id (move within the group/non-group).